src/components/sidebar/routePathView/routePathListTab/RoutePathListLink.tsx

```diff
@@ -21,11 +23,87 @@ interface IRoutePathListLinkProps {
     codeListStore?: CodeListStore;
     routePathLink: IRoutePathLink;
     reference: React.RefObject<HTMLDivElement>;
+    isEditingDisabled: boolean;
+}
+
+interface IRoutePathListLinkState {
+    isLoading: boolean; // not currently in use, declared because ViewFormBase needs this
+    invalidPropertiesMap: object;
+    isEditingDisabled: boolean; // not currently in use, declared because ViewFormBase needs this
 }
 
 @inject('routePathStore', 'codeListStore')
 @observer
-class RoutePathListLink extends React.Component<IRoutePathListLinkProps> {
+class RoutePathListLink extends ViewFormBase<
+    IRoutePathListLinkProps,
+    IRoutePathListLinkState
+> {
+    constructor(props: IRoutePathListLinkProps) {
+        super(props);
+        this.state = {
+            isLoading: false,
+            invalidPropertiesMap: {},
+            isEditingDisabled: false
+        };
+    }
+
+    componentDidMount() {
+        this.validateLink();
+    }
+
+    componentDidUpdate(prevProps: IRoutePathListLinkProps) {
+        if (
+            prevProps.isEditingDisabled !== this.props.isEditingDisabled &&
+            !this.props.isEditingDisabled
+        ) {
+            this.validateLink();
+        }
+    }
+
+    private validateLink = () => {
+        this.validateAllProperties(
+            routePathLinkValidationModel,
+            this.props.routePathLink
+        );
+        const isLinkFormValid = this.isFormValid();
+        const orderNumber = this.props.routePathLink.orderNumber;
+        this.props.routePathStore!.setLinkFormValidity(
+            orderNumber,
+            isLinkFormValid
+        );
+    };
+
+    private onCheckboxChange = (property: string, value: boolean) => () => {
+        const orderNumber = this.props.routePathLink.orderNumber;
+        this.props.routePathStore!.updateRoutePathLinkProperty(
+            orderNumber,
+            property,
+            !value
+        );
+    };
+
+    private onRoutePathLinkPropertyChange = (property: string) => (
+        value: any
+    ) => {
+        const _value = value === '' ? null : value;
```

Comment:
I assume this is done in order to prevent empty strings in the database. If it is indeed for that reason, then I don't think this is the place where that logic should happen.

Reply:
I guess this was because I thought startNodeColumnNumber didn't quite work when it was ''. Not it works, maybe because of other bug fixes --> i removed this line